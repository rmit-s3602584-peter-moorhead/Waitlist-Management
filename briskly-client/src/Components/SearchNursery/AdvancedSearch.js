import React, {Component} from 'react'
import "../../ComponentStyles.css"

class AdvancedSearch extends Component{

    /*

        This component show the Advanced Search options for the user to select from

    */

    render(){
        return(
        
            <div id="advancedSearch">
               
                <div class="form-group">
                    <div class="form-check"><input class="form-check-input" type="checkbox" name="ChildCareFeedReduction" value="yes" id="formCheck-2"/><label class="form-check-label" for="formCheck-2">Your center is a member of the Child Care Fee Reduction Initiative</label></div>
                </div>

                 <div class="form-group">
                    <div class="form-check"><input class="form-check-input" type="checkbox" name="ECE" value="yes" id="formCheck-3"/><label class="form-check-label" for="formCheck-3">ECE (Early Childhood Education) Certification</label></div>
                </div>

                <div class="form-group">
                    <div class="form-check"><input class="form-check-input" type="checkbox" name="ELF" value="yes" id="formCheck-4"/><label class="form-check-label" for="formCheck-4">ELF (Early Learning Framework) Programming</label></div>
                </div>


                <div class="form-group">
                    <div><label>Facility type:</label>
                        <div class="form-check"><input name="FacilityType" class="form-check-input" name="faciltyType" type="radio" value="Licensed Family" id="formCheck-9"/><label class="form-check-label" for="formCheck-9">Licensed Family</label></div>
                        <div class="form-check"><input name="FacilityType" class="form-check-input" name="faciltyType" type="radio" value="Licensed Group" id="formCheck-10"/><label class="form-check-label" for="formCheck-10">Licensed Group</label></div>
                    </div>
                </div>


                <div class="form-group"><label>Program(s):</label>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-11" name="programs1" value="Under 36 Months"/><label class="form-check-label" for="formCheck-11">Under 36 Months</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-14" name="programs2" value="3 years to Kindergarten"/><label class="form-check-label" for="formCheck-14">3 years to Kindergarten</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-13" name="programs3" value="Licensed Preschool"/><label class="form-check-label" for="formCheck-13">Licensed Preschool</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-12" name="programs4" value="School Age: Grade 1 to Age 12"/><label class="form-check-label" for="formCheck-12">School Age: Grade 1 to Age 12</label></div>
                </div>


                <div class="form-group"><label>House of Operation:</label>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-15" name="HouseOfOperation1" value="Weekdays"/><label class="form-check-label" for="formCheck-15">Weekdays</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-20" name="HouseOfOperation2" value="Weekends"/><label class="form-check-label" for="formCheck-20">Weekends</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-19" name="HouseOfOperation3" value="Statutory Holidays"/><label class="form-check-label" for="formCheck-19">Statutory Holidays</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-18" name="HouseOfOperation4" value="Overnight"/><label class="form-check-label" for="formCheck-18">Overnight</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-17" name="HouseOfOperation5" value="Weekdays Before 6am"/><label class="form-check-label" for="formCheck-17">Weekdays Before 6am</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-16" name="HouseOfOperation6" value="Weekdays After 7pm"/><label class="form-check-label" for="formCheck-16">Weekdays After 7pm</label></div>
                </div>

                <div class="form-group"><label>Meal Services:</label>
                    <div class="form-check"><input class="form-check-input" value="Extra Fee" name="MealServices" type="radio" id="formCheck-21"/><label class="form-check-label" for="formCheck-21">Extra Fee</label></div>
                    <div class="form-check"><input class="form-check-input" value="Included in fee" name="MealServices" type="radio" id="formCheck-23"/><label class="form-check-label" for="formCheck-23">Included in fee</label></div>
                    <div class="form-check"><input class="form-check-input" value="Not offered" name="MealServices" type="radio" id="formCheck-22"/><label class="form-check-label" for="formCheck-22">Not offered</label></div>
                </div>

                <div class="form-group"><label>Pick-up Service:</label>
                    <div class="form-check"><input class="form-check-input" value="Extra Fee" name="pickup" type="radio" id="formCheck-24"/><label class="form-check-label" for="formCheck-24">Extra Fee</label></div>
                    <div class="form-check"><input class="form-check-input" value="Included in fee" name="pickup" type="radio" id="formCheck-26"/><label class="form-check-label" for="formCheck-26">Included in fee</label></div>
                    <div class="form-check"><input class="form-check-input" value="Not offered" name="pickup" type="radio" id="formCheck-25"/><label class="form-check-label" for="formCheck-25">Not offered</label></div>
                </div>
                <div class="form-group"><label>Language:</label>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-502" name="Language6" value="English" /><label class="form-check-label" for="formCheck-502">English</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-27" name="Language1" value="Cantonese" /><label class="form-check-label" for="formCheck-27">Cantonese</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-31" name="Language2" value="French"/><label class="form-check-label" for="formCheck-31">French</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-30" name="Language3" value="Mandarin"/><label class="form-check-label" for="formCheck-30">Mandarin</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-29" name="Language4" value="Punjabi"/><label class="form-check-label" for="formCheck-29">Punjabi</label></div>
                    <div class="form-check"><input class="form-check-input" type="checkbox" id="formCheck-28" name="Language5" value="Spanish"/><label class="form-check-label" for="formCheck-28">Spanish</label></div>
                </div>
                <div class="form-group"><label>Aboriginal Programming:</label>
                    <div class="form-check"><input class="form-check-input" value="yes" name="Aboriginal" type="radio" id="formCheck-5"/><label class="form-check-label" for="formCheck-5">Yes</label></div>
                    <div class="form-check"><input class="form-check-input" value="no" name="Aboriginal" type="radio" id="formCheck-6"/><label class="form-check-label" for="formCheck-6">No</label></div>
                </div>
                <div class="form-group"><label>Special Needs</label>
                    <div class="form-check"><input class="form-check-input" value="yes" name="specialNeeds" type="radio" id="formCheck-7"/><label class="form-check-label" for="formCheck-7">Yes</label></div>
                    <div class="form-check"><input class="form-check-input" value="no" name="specialNeeds" type="radio" id="formCheck-8"/><label class="form-check-label" for="formCheck-8">No</label></div>
                </div>
            

              
            </div>
          



        )
    }

}

export default AdvancedSearch