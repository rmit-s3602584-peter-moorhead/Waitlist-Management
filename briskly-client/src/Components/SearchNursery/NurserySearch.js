import React, {Component} from 'react'


import CountryStateCity from "../CountryStateCity.js"
import Axios from 'axios'
import AdvancedSearch from "./AdvancedSearch.js"
import "../../ComponentStyles.css"


class NurserySearch extends Component{
    
    /*

    This component is used to hold the nursery search page that will be used
    to fill in some data and then the user will hit search and then will be taken
    to another component to show the results.

    */
    
    constructor(props){
        super(props)

        //Storing some variables in the component state
        this.state = {
            siblings: "",
            showAdvancedSearch: false,
            advancedButtonTitle: "Advanced Search",
            result: [],
            finishedSearch: false
        }

        //in case there is a previos search, remove it
        sessionStorage.removeItem("nurserySearchResults")
    }


    submit = (e) => {
        //preventing the default submission behavior
        //because we wanna do data processing first
        e.preventDefault();

        //storing the event target
        let target = e.target;
        //what will be sent to the backend
        let formData = {};
        /**
         * 
         * Note::::
         * 
         * I'm using here the name property of HTML components to retreive their values
         * 
         * */

        for (var i = 0; i < target.length; i++) {
            
            //if the current target element is a checkbox or a radio button
            if(target.elements[i].type == "checkbox" || target.elements[i].type == "radio"){
            
                //if the radio or checkbox is selected    
                if(target[i].checked == true){
                    //add the element to the formData dictionary
                    formData[target.elements[i].getAttribute("name")] = target[i].value;
                }

            }else{
                //if it is not a checkbox or radio then just simply add it to
                //the formData
                formData[target.elements[i].getAttribute("name")] = target[i].value;
            }
        }

        //filtering out the data for the backend in case the user does
        //not select a country/city/state
        if(formData["country"] == "Select Country"){
            formData["country"] = undefined
        }
        if(formData["state"] == "Select State/Province"){
            formData["state"] = undefined
        }
        if(formData["city"] == "Select City"){
            formData["city"] = undefined
        }
        if(formData["facilityType"] == "Licensed Family"){
            formData["LicensedFamily"] = 1
        }
        if(formData["facilityType"] == "Licensed Group"){
            formData["LicensedGroup"] = 1
        }
        if(formData["ChildCareFeedReduction"] != undefined){
            if(formData["ChildCareFeedReduction"] == "yes"){
                formData["ChildCareFeedReduction"] = 1
            }else{
                formData["ChildCareFeedReduction"] = 0
            }
        }
        if(formData["ECE"] != undefined){
            if(formData["ECE"] == "yes"){
                formData["ECE"] = 1
            }else{
                formData["ECE"] = 0
            }
        }
        
        if(formData["ELF"] != undefined){
            if(formData["ELF"] == "yes"){
                formData["ELF"] = 1
            }else{
                formData["ELF"] = 0
            }
        }

        if(formData["specialNeeds"] != undefined){
            if(formData["specialNeeds"] == "yes"){
                formData["specialNeeds"] = 1
            }else{
                formData["specialNeeds"] = 0
            }
        }
        
        //After the form is filled now we send the data to the backend using the method below.
        //it is done in a seperate method to use the 'async' feature to be able to wait for a response
        //more effeciently
        this.getNurseries(formData);
        
    }

    async getNurseries(formData){

        //storing all the nurseries
        var nurseries = []
        //sending a post request with the formData
        const response = await Axios.post('http://localhost:9000/searchNursery', {...formData})
        
        
        //storing the nurseries in the nurseries array
        for(var x in response['data']){
            nurseries.push(response['data'][x])
        }

        //setting the state with the nurseries array
        this.setState({
            result:nurseries
        })

        //setting the 'finishedSearch' variable to true
        this.setState({
            finishedSearch: true
        })

        // if search is finished then we proceed to /nsr page
        //which will show us the results of our search
        if(this.state.finishedSearch){
            this.props.history.push({
                pathname:"/nsr",
                result: this.state.result
            })
        }


    }

    siblingsCheck = (e) => {
        //adding the sibling value to the siblings state variable
        this.setState({
            siblings: e.target.value
        })
    }

    advancedSearch = (e) =>{
        
        //when advance search button is clicked
        //change the showAdvancedSearch value
        this.state.showAdvancedSearch = !this.state.showAdvancedSearch
        //it is done so that is the advanced search is shown
        //then hide it and vice versa

        //here i'm setting the title of the button
        if(this.state.showAdvancedSearch == true){
            this.state.advancedButtonTitle = "Hide Advanced Search"
        }else{
            this.state.advancedButtonTitle = "Advanced Search"
        }

        //updating the component on the screen
        this.forceUpdate()
    }



    render(){

        /**
         * 
         * The commented code below will be useful for whoever will takeover the project
         * 
         * The client asked if they could have the parent find a specific nursery for the child in case
         * a sibling is already enrolled in that nursery. But we haven't been able to implement that because
         * of time constraints. So this for the person taking charge of this part. It is already done for the
         * front-end you just need to uncomment it here and in the HTML below and send it to the backend and 
         * populate the drop-down (populating is what you need to write), the rest is completed
         * 
         */

        // let selectNurseryForSibling
        // if(this.state.siblings == "yes"){
        //     //get all the nurseries in the city that the user selected.
            
        //     selectNurseryForSibling = (
        //         <div style={{display: 'flex', flexDirection: 'row'}}>
        //         <div class="countryStateCityLabel" ><label style={{fontSize: 'calc(0.5em + 1vmin)'}}>Select sibling nursery:</label></div>
              
        //         <div style={{width: '50%'}}>
        //             <select class="form-control countryStateCityDropdown" name="siblingNursery">
        //                 <option  style={{backgroundColor: 'rgb(219,222,225)',color: 'rgb(0,0,0)',margin: 0}} data-toggle="dropdown" aria-expanded="false">Select Nursery&nbsp;</option>
        //                         {
        //                         //   this.state.states.map((item, i) => <option name="siblingNursery" value={this.state.states[i]} key={i}>{this.state.states[i]}</option>) 
        //                         }                          
        //         </select>

        //     </div>
        // </div>
        //     )

        // }

        let advancedSearch

        if(this.state.showAdvancedSearch == true){
            advancedSearch = (
               //showing the AdvancedSearch component if the state variable (showAdvancedSearch) is true
                <AdvancedSearch/>
               
            )
        }else{
            advancedSearch = null
        }

        
        return(
            
           

            <div class="container">
                <form method='post' onSubmit={this.submit} >
                    <div style={{textAlign: "center", marginBottom: '3vh', marginTop: '3vh'}}>
                        <h2> This form will help you reach the nursery that you are looking for</h2>
                    </div>
                <label class="title">Geographical Area</label>
                 <div ><CountryStateCity keyToUse={0} country="country" state="state" city="city"/></div>
        
                 {/* <div class="form-group"><label>Does the child have a sibling in any other daycare in the same city?</label>
                    <div class="form-check"><input class="form-check-input" value="yes" name="siblings" onClick={this.siblingsCheck} type="radio" id="formCheck-7"/><label class="form-check-label" for="formCheck-5">Yes</label></div>
                    <div class="form-check"><input class="form-check-input" value="no" name="siblings" onClick={this.siblingsCheck} type="radio" id="formCheck-8"/><label class="form-check-label" for="formCheck-5">No</label></div>
                    {
                        selectNurseryForSibling
                    }

                </div> */}



                    <button type="button" id="advancedSearchButton" onClick={this.advancedSearch}>{this.state.advancedButtonTitle}</button>

                    {
                        //the variable that will either hold the advanced search component to render it
                        //or not hold anything and not show it
                        advancedSearch
                    }

                    <div class="form-group" style={{width: '100%' ,textAlign: "center"}}>
                              
                        <button class="btn search" type="submit">Search</button>
                    </div> 
                    

                 </form>



        </div>
    



        )
    }


}

export default NurserySearch