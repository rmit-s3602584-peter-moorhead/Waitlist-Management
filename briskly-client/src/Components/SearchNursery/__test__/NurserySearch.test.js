import React from 'react'
import ReactDOM from 'react-dom'
import NurserySearch from '../NurserySearch'
import AdvancedSearch from '../AdvancedSearch'
import "../../../setupTests"
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from "react-test-renderer"

import { shallow, mount } from 'enzyme';
import ReactTestUtils, { act } from 'react-dom/test-utils';

afterEach(cleanup)


it("renders without crashing", () =>{
    //testing it the page actually renders or not
    const div = document.createElement("div");
    ReactDOM.render(<NurserySearch></NurserySearch>, div)
})


it("matches snapshot", () => {
    //creating a snapshot test
    const tree = renderer.create(<NurserySearch></NurserySearch>).toJSON()
    expect(tree).toMatchSnapshot()
})
//advancedSearch
//advancedSearchButton
it("Checking to see if the advanced search appears or not", () => {
    const wrapper = mount(<NurserySearch/>)
    const btn = wrapper.find('#advancedSearchButton')
    btn.simulate('click')
    wrapper.update()

    expect(wrapper.find('#advancedSearchButton').text()).toBe("Hide Advanced Search")
})

it("Checking to see if the advanced search hides after clicking it again or not", () => {
    const wrapper = mount(<NurserySearch/>)
    const btn = wrapper.find('#advancedSearchButton')
    btn.simulate('click')
    wrapper.update()
    btn.simulate('click')
    wrapper.update()

    expect(wrapper.find('#advancedSearchButton').text()).toBe("Advanced Search")
})

it("Checking to see if the advanced search fields appear or not", () => {
    const wrapper = mount(<NurserySearch/>)
    const btn = wrapper.find('#advancedSearchButton')
    btn.simulate('click')
    wrapper.update()

    expect(wrapper.find('#advancedSearch')).toHaveLength(1)
})

it("Checking to see if the advanced search fields disappears after clicking the button or not", () => {
    const wrapper = mount(<NurserySearch/>)
    const btn = wrapper.find('#advancedSearchButton')
    btn.simulate('click')
    wrapper.update()
    btn.simulate('click')
    wrapper.update()

    expect(wrapper.find('#advancedSearch')).toHaveLength(0)
})