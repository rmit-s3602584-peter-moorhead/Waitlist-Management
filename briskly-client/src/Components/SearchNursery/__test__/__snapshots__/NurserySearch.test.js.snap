// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="container"
>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <div
      style={
        Object {
          "marginBottom": "3vh",
          "marginTop": "3vh",
          "textAlign": "center",
        }
      }
    >
      <h2>
         This form will help you reach the nursery that you are looking for
      </h2>
    </div>
    <label
      class="title"
    >
      Geographical Area
    </label>
    <div>
      <div>
        <div
          class="form-group"
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              class="countryStateCityLabel"
            >
              <label
                style={
                  Object {
                    "fontSize": "calc(0.5em + 1vmin)",
                  }
                }
              >
                Country:
              </label>
            </div>
            <div
              style={
                Object {
                  "width": "50%",
                }
              }
            >
              <select
                class="form-control countryStateCityDropdown"
                id="country"
                name="country"
                onChange={[Function]}
              >
                <option
                  aria-expanded="false"
                  data-toggle="dropdown"
                  style={
                    Object {
                      "backgroundColor": "rgb(219,222,225)",
                      "color": "rgb(0,0,0)",
                      "margin": 0,
                    }
                  }
                >
                  Select Country
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              class="countryStateCityLabel"
            >
              <label
                style={
                  Object {
                    "fontSize": "calc(0.5em + 1vmin)",
                  }
                }
              >
                State/Province:
              </label>
            </div>
            <div
              style={
                Object {
                  "width": "50%",
                }
              }
            >
              <select
                class="form-control countryStateCityDropdown"
                name="state"
                onChange={[Function]}
              >
                <option
                  aria-expanded="false"
                  data-toggle="dropdown"
                  style={
                    Object {
                      "backgroundColor": "rgb(219,222,225)",
                      "color": "rgb(0,0,0)",
                      "margin": 0,
                    }
                  }
                >
                  Select State/Province
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              class="countryStateCityLabel"
            >
              <label
                style={
                  Object {
                    "fontSize": "calc(0.5em + 1vmin)",
                    "margin": "0px",
                  }
                }
              >
                City:
              </label>
            </div>
            <div
              style={
                Object {
                  "width": "50%",
                }
              }
            >
              <select
                class="form-control countryStateCityDropdown"
                name="city"
              >
                <option
                  aria-expanded="false"
                  data-toggle="dropdown"
                  style={
                    Object {
                      "backgroundColor": "rgb(219,222,225)",
                      "color": "rgb(0,0,0)",
                      "margin": 0,
                    }
                  }
                >
                  Select City
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      id="advancedSearchButton"
      onClick={[Function]}
      type="button"
    >
      Advanced Search
    </button>
    <div
      class="form-group"
      style={
        Object {
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      <button
        class="btn search"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
